♻️ Simplify code in `fastapi/routing.py`

bokshitsky opened the pull request:
TLDR: just simplify internal logic and code a bit, doesn't change framework behaviour


split in in 4 commits:
1) remove unused variable initialization `errors: List[Any] = []`


2) simplify code: remove nested code, using early exit pattern

```
if errors:
  raise Exception()
```


3) instead of:
```
current_status_code = (
    status_code if status_code else solved_result.response.status_code
)
if current_status_code is not None:
    response_args["status_code"] = current_status_code
if solved_result.response.status_code:
    response_args["status_code"] = solved_result.response.status_code
```

write:
```
if solved_result.response.status_code:
    response_args["status_code"] = solved_result.response.status_code
elif status_code is not None:
    response_args["status_code"] = status_code
```

if remains exactly the old behaviour, but 
a) looks more clear (right now looks like `status_code if status_code else solved_result.response.status_code` is default value, but actually in the end default value is `solved_result.response.status_code`
b) it doesn't make any speed effect, but new version use less comparison and assignment operations


4) remove non used variable `response` initialization

bokshitsky committed:
🔨 refactoring in fastapi/routing

bokshitsky commented on fastapi/routing.py:
```diff
@@ -386,52 +393,41 @@ async def app(request: Request) -> Response:
+            raise RequestValidationError(_normalize_errors(errors), body=body)
 
-        # Return response
-        assert response
```
`assert response` was used to avoid forget response assignment inside if-else-branches, not required is proposed changes

YuriiMotov reviewed (approved):
LGTM!

With these changes code is more readable. Changes don't affect final result

@bokshitsky, thank you!

bokshitsky commented:
@YuriiMotov, I rebased code and fixed conficts, 
see some failing checks, not sure what's the reason, probably need rerun them, but I don't have rerun button in interface, do you have?
<img width="553" height="317" alt="image" src="https://github.com/user-attachments/assets/328c3541-d97b-4e7f-99fb-16f9a48cc606" />

@tiangolo, may you please look, if this PR is acceptable and merge if everything ok?
I believe, since some merge-conflicts occured and the logic inside actually changes in new commits, the code is "live" and code simplification from that PR make sense

pre-commit-ci-lite[bot] committed:
🎨 Auto format

tiangolo commented:
Thanks for the interest and work on this! ☕

I want to implement a big refactor in how routers work.

Adding this change means exposing the possibility that it breaks someone's use case somewhere, and that risk is not worth now given there's that additional work coming.

I have not wanted to close PRs that target the same or similar problems before I have an alternative, but then I have left many PRs open for too long, which is not useful or clear either. Sorry for that.

So, I'll close this one for now to keep things clean.

Thank you! 🍰
